static/src/javascripts/projects/common/modules/commercial/create-ad-slot.js

```diff
@@ -1,131 +1,153 @@
+                ),
+                tablet: compile(
+                    adSizes.outOfPage,
+                    adSizes.mpu,
+                    config.switches.fluidAdverts ? adSizes.fluid : null
+                )
             }
         },
         inline: {
             sizeMappings: {
-                mobile:             '1,1|300,250',
-                'mobile-landscape': '1,1|300,250',
-                tablet:             '1,1|300,250'
+                mobile: compile(adSizes.outOfPage, adSizes.mpu)
             }
         },
         mostpop: {
             sizeMappings: {
-                mobile:             '1,1|300,250',
-                'mobile-landscape': '1,1|300,250',
-                tablet:             '1,1|300,250'
+                mobile: compile(adSizes.outOfPage, adSizes.mpu)
             }
         },
         'merchandising-high': {
             label: false,
             refresh: false,
             sizeMappings: {
-                mobile: '1,1|88,87'
+                mobile: compile(adSizes.outOfPage, adSizes.merchandising)
             }
         },
         spbadge: {
             label: false,
             refresh: false,
             sizeMappings: {
-                mobile: '1,1|140,90'
+                mobile: compile(adSizes.outOfPage, adSizes.badge)
             }
         },
         adbadge: {
             label: false,
             refresh: false,
             sizeMappings: {
-                mobile: '1,1|140,90'
+                mobile: compile(adSizes.outOfPage, adSizes.badge)
             }
         },
         fobadge: {
             label: false,
             refresh: false,
             sizeMappings: {
-                mobile: '1,1|140,90'
+                mobile: compile(adSizes.outOfPage, adSizes.badge)
             }
         },
         comments: {
             sizeMappings: {
-                mobile:  '1,1|300,250'
+                mobile:  compile(adSizes.outOfPage, adSizes.badge)
             }
         },
         'top-above-nav': {
             sizeMappings: {
-                desktop: '1,1|88,70|728,90|940,230|900,250|970,250|' + fabricTopSlot
+                desktop: compile(
+                    adSizes.outOfPage,
+                    adSizes.fluid250,
+                    adSizes.leaderboard,
+                    adSizes.cascase,
+                    adSizes.superHeader,
+                    adSizes.billboard,
+                    adSizes.fabric
+                )
             }
         }
     };
 
-    function createAdSlotElement(name, attrs, classes) {
-        return $(document.createElement('div'))
-            .attr({
-                'id': 'dfp-ad--' + name,
-                'data-link-name': 'ad slot ' + name,
-                'data-test-id': 'ad-slot-' + name,
-                'data-name': name
-            })
-            .attr(attrs)
-            .addClass('js-ad-slot ad-slot ad-slot--dfp ' + classes.join(' '));
+    function compile(size1) {
```

Comment:
Nope, it's just that the signature of `compile` is `size1: AdSize [,
AdSize]*`. I can change it if it isn't clear enough.

On Tue, 14 Jun 2016 at 17:43 Richard Nguyen <notifications@github.com>
wrote:

> In
> static/src/javascripts/projects/common/modules/commercial/create-ad-slot.js
> <https://github.com/guardian/frontend/pull/13291#discussion_r67008353>:
>
> >              }
> >          }
> >      };
> >
> > -    function createAdSlotElement(name, attrs, classes) {
> > -        return $(document.createElement('div'))
> > -            .attr({
> > -                'id': 'dfp-ad--' + name,
> > -                'data-link-name': 'ad slot ' + name,
> > -                'data-test-id': 'ad-slot-' + name,
> > -                'data-name': name
> > -            })
> > -            .attr(attrs)
> > -            .addClass('js-ad-slot ad-slot ad-slot--dfp ' + classes.join(' '));
> > +    function compile(size1) {
>
> size1 instead of size? is there something else in scope?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/guardian/frontend/pull/13291/files/efc019abdb440a6ceb90f2e9508ebd1e1222d0f6#r67008353>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAmc2L2ER5hPokzrlI_qdA88fXCmMOzBks5qLtoWgaJpZM4I1ei9>
> .
>

-- 

------------------------------
This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396